### PR TITLE
[1.19.x] Guard player and living entity pre-render events against unbalanced PoseStack modification

### LINKED ISSUE #9118: RenderLivingEvent Posestacks
**Minecraft Version:** 1.18.2 but should also go for 1.19 since the event didn't change

**Forge Version:** 40.1.73

**Steps to Reproduce:**
See description since this is hard to reproduce.

**Description of issue:**

This is an issue that occurs only if multiple mobs are present but only under very specific circumstances. 
At least 2 mods need to subscribe to `RenderLivingEvent.Pre`, the mod with the higher priority modifies (pushes) the posestack with the intention on passing the stack on to the normal renderer. The mod with lesser priority cancels the event and does NOT call `RenderLivingEvent.Post`.

Because the posestack was pushed in the first mod but since `RenderLivingEvent.Post` is not called the stack never gets popped and thus creates a `Pose stack not empty` error down the line. Additionally this error will be obscure to both users and modders: 
[https://gist.github.com/Flemmli97/416415eaf4ee973db98a170c235a9ac4](https://gist.github.com/Flemmli97/416415eaf4ee973db98a170c235a9ac4)

Forge should add something that takes this into consideration. Maybe when the event is cancelled call `RenderLivingEvent.Post` or add java docs to it (though then it means depending on modders to pay attention to it). The first one also might be debatable.

While this might seem a mod issue at first i reported it here since this is really obscure and people might not notice this problem.

### PR DESCRIPTION
**Pull request by XFactHD**
This PR implements a guard on `RenderPlayerEvent.Pre` and `RenderLivingEvent.Pre` to prevent the game from crashing when a higher priority event handler pushes the `PoseStack` given by the event and a lower priority event handler then cancels the event, preventing the owner of the higher priority handler from popping the stack again in the respective post event.

Fixes #9118

### CODE REVIEW

File: src/test/java/net/minecraftforge/debug/client/rendering/RenderLivingEventTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package net.minecraftforge.debug.client.rendering;
```
Comment (ChampionAsh5357) [inline src/test/java/net/minecraftforge/debug/client/rendering/RenderLivingEventTest.java:6]: Missing license header.

**Review by ChampionAsh5357 — CHANGES_REQUESTED** [review]
As a future note, I believe that the event cancelation should probably be handled before any of these transformations get applied. So, it would probably make sense to enqueue the rendering work into a runnable and then check to make sure the event wasn't canceled before executing the logic. Either that or separate them into a registry and the cancelation event, though that would be two systems.

TL;DR, this is fine for the current solution. However, I believe we should probably redesign this event if this is the purpose it is used for.

**Comment by XFactHD** [comment]
Handling modifications with an enqueued runnable would probably be the cleanest solution for the future but some kind of guard against problematic modifications to the `PoseStack` with an additional error message should be retained even if such a solution is implemented to allow proper debugging when someone uses the current approach.

**Review by ChampionAsh5357 — APPROVED** [review]
Seems fine.

File: src/main/java/net/minecraftforge/client/ForgeHooksClient.java [hunk 2]
```diff
@@ -1257,4 +1262,30 @@ public static void initClientHooks(Minecraft mc, ReloadableResourceManager resou
...
+        if (MinecraftForge.EVENT_BUS.post(event))
+        {
+            while (poseStack.size() > oldSize)
+            {
+                poseStack.popPose();
+            }
+            return true;
+        }
+        return false;
+	}
```
Comment (ChampionAsh5357) [inline src/main/java/net/minecraftforge/client/ForgeHooksClient.java]: Tab instead of whitespace

**Review by ChampionAsh5357 — CHANGES_REQUESTED** [review]

**Review by ChampionAsh5357 — APPROVED** [review]
